### PR TITLE
4118 -  update product subscription task implementation

### PR DESCRIPTION
**Pull request by karthik-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/4118

*Description of changes:*

- Update task to handle update product subscriptions
- Pagination for fetch tasks
- Test cases

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-api/src/auth_api/services/task.py [hunk 1]
```diff
@@ -101,6 +103,14 @@ def update_relationship(self, task_relationship_status: str, user_name: str):
             org_id = task_model.relationship_id
             self.update_org(is_approved=is_approved, org_id=org_id,
                             user_name=user_name)
+
+        elif task_model.relationship_type == TaskRelationshipType.PRODUCT.value:
+            # Update Product relationship
+            product_subscription_id = task_model.relationship_id
+            print(product_subscription_id)
```
Comment (sumesh-aot) [inline auth-api/src/auth_api/services/task.py:110]: remove print

File: auth-api/src/auth_api/services/task.py [hunk 2]
```diff
@@ -130,14 +140,38 @@ def update_org(is_approved: bool, org_id: int, user_name: str):
         current_app.logger.debug('>update_task_org ')
 
     @staticmethod
-    def fetch_tasks(task_type: str, task_status: str):
+    def update_product_subscription(is_approved: bool, product_subscription_id: int):
+        """Review Product Subscription."""
+        current_app.logger.debug('<update_task_product ')
+        # Approve/Reject Product subscription
+        product_subscription: ProductSubscriptionModel = ProductSubscriptionModel.find_by_id(product_subscription_id)
+        print(product_subscription.id)
```
Comment (sumesh-aot) [inline auth-api/src/auth_api/services/task.py:148]: print

**Review by sumesh-aot — APPROVED** [review]

File: auth-api/src/auth_api/services/products.py [hunk 3]
```diff
@@ -85,6 +88,18 @@ def create_product_subscription(org_id, subscription_data: Dict[str, Any], is_ne
...
                                                                 status_code=product_model.default_subscription_status)\
                     .flush()
+                # create a staff review task for this product subscription
+                user = UserModel.find_by_jwt_token(token=token_info)
+                task_info = {'name': product_model.description,
+                             'relationshipId': product_subscription.id,
+                             'relatedTo': user.id,
+                             'dateSubmitted': datetime.today(),
+                             'relationshipType': TaskRelationshipType.PRODUCT.value,
+                             'type': TaskType.PENDING_STAFF_REVIEW.value,
```
Comment (saravanpa-aot) [inline auth-api/src/auth_api/services/products.py:98]: I think we have a way to find the status ; something like default status of a product;

Comment (karthik-aot) [reply auth-api/src/auth_api/services/products.py:98]: As per our discussion. need to discuss more shall do at a later PR

**Pull request merged**